tracing-subscriber/src/filter/env/directive.rs

```diff
@@ -1003,6 +1029,34 @@ mod test {
         assert_eq!(dirs[1].in_span, None);
     }
 
+    #[test]
+    fn parse_directives_global_bare_warn_lc() {
+        // test parse_directives with no crate, in isolation, all lowercase
+        let dirs = parse_directives("warn");
+        assert_eq!(dirs.len(), 1, "\nparsed: {:#?}", dirs);
+        assert_eq!(dirs[0].target, None);
+        assert_eq!(dirs[0].level, LevelFilter::WARN);
+        assert_eq!(dirs[0].in_span, None);
+    }
+
+    #[test]
+    fn parse_directives_global_bare_warn_uc() {
+        // test parse_directives with no crate, in isolation, all uppercase
+        let dirs = parse_directives("WARN");
+        assert_eq!(dirs[0].target, None);
+        assert_eq!(dirs[0].level, LevelFilter::WARN);
+        assert_eq!(dirs[0].in_span, None);
+    }
+
+    #[test]
+    fn parse_directives_global_bare_warn_mixed() {
+        // test parse_directives with no crate, in isolation, mixed case
+        let dirs = parse_directives("wArN");
+        assert_eq!(dirs[0].target, None);
+        assert_eq!(dirs[0].level, LevelFilter::WARN);
+        assert_eq!(dirs[0].in_span, None);
+    }
```

Comment:
Since this PR adds so many new test cases that are all pretty equivalent, I wonder if it's worth trying to factor out some of the repeated code. For example, we could do something like:

```suggestion
    fn test_parse_bare_level(directive: &str, level: LevelFilter) {
        let dirs = parse_directives(directive);
        assert_eq!(dirs.len(), 1, "\nparsed: {:#?}", dirs);
        assert_eq!(dirs[0].target, None);
        assert_eq!(dirs[0].level, LevelFilter::WARN);
        assert_eq!(dirs[0].in_span, None);
    }
  
    #[test]
    fn parse_directives_global_bare_warn_lc() {
        // test parse_directives with no crate, in isolation, all lowercase
        test_parse_bare_level("warn", LevelFilter::WARN);
    }

    #[test]
    fn parse_directives_global_bare_warn_uc() {
        // test parse_directives with no crate, in isolation, all uppercase
        test_parse_bare_level("WARN", LevelFilter::WARN);
    }

    #[test]
    fn parse_directives_global_bare_warn_mixed() {
        // test parse_directives with no crate, in isolation, mixed case
        test_parse_bare_level("wArN", LevelFilter::WARN);
    }
```

If you don't think this is worth the effort, I'm happy to merge this PR either way — just a thought!

Reply:
> Since this PR adds so many new test cases that are all pretty equivalent, I wonder if it's worth trying to factor out some of the repeated code. For example, we could do something like:
> 
> If you don't think this is worth the effort, I'm happy to merge this PR either way — just a thought!

Thanks for the suggestion; that's clearly better. I'll incorporate it and rebase.

Reply:
I incorporated this idea into my push from earlier tonight.